### ISSUE TITLE
"Keep" and "Discard" are confusing options in the "unsaved changes" dialog.

### ISSUE THREAD

**Issue by Raketemensch** [SoftFever/OrcaSlicer#4422]
### Is there an existing issue for this feature request?

- [X] I have searched the existing issues

### Is your feature request related to a problem?

The button choices in this dialog are confusing.

The text says, "Would you like to keep these changed settings (new value) after switching presets?"

The table in the dialog shows "Old Value" and "New Value," and the button choices are "Transfer" and "Discard," which don't make sense in this context.

If I hit "Discard" am I getting rid of the old settings? The new changes? It's not clear to the user.

The buttons should say "Use New" and "Keep Old" or something more explicit.

### Which printers will be beneficial to this feature?

All

### Describe the solution you'd like

The buttons should say "Use New" and "Keep Old" or something more explicit.

They should leverage the text that is already in the dialog.

**Comment by mtmello2003** [comment]
I totally agree.

**Comment by Eldenroot** [comment]
Yes

**Comment by xTremePower** [comment]
totally agree ..
It made me errors many times ... and saved what I don't want to save ...

**Comment by nmcandrews** [comment]
100% agree.

**Comment by SoftFever** [comment]
It should be more clear/consistent in 2.1.0 beta now
![image](https://github.com/SoftFever/OrcaSlicer/assets/103989404/cffe8a8e-f738-4979-bacd-70026f495175)

**Comment by ShayGus** [comment]
This is sooooooo confusing.
What am I transfering new to old, old to new?
What am I discarding of? The new value, the old one.
I never quite sure what's going on the re when I press either.
As a user I don't have to need to go and look at the code to understand a simple dialog.
My UX is that I totally and absolutelly confused by this dialog.

**Comment by sdylewski** [comment]
I'd really like some more help on this screen also.  My wife and kid and I all use the same slicer and we're all just playing roulette every time we click one of the buttons because we don't understand it.  It shouldn't be that way.  This is just not obvious to me what each option does.

**Comment by PitBull440** [comment]
I agree completely.  It's almost like this was written as a cruel joke to confuse people purposely.    

How about putting a checkbox above or below each column so we can clearly select which settings we are accepting?

And what does the "Save" button do?

**Comment by adobni** [comment]
Current options are confusing. Am I overriding the defaults? or is it the project? what is transferred to what?

**Comment by wickedlemon2** [comment]
> Please read the description in the dialog, it's pretty clear already.
> 
> This dialog popup when you switch presets and there are unsaved changes. So these options are: **Transfer** (changes to new preset) **Discard** (changes) **Save** changes into current preset then switch to new preset

If it's pretty clear already why did you need to append additional descriptions in your post?

**Comment by TheSiskoDax** [comment]
> Please read the description in the dialog, it's pretty clear already.
> 
> This dialog popup when you switch presets and there are unsaved changes. So these options are: **Transfer** (changes to new preset) **Discard** (changes) **Save** changes into current preset then switch to new preset

This is the question: If I am downloading a .3mf from makersworld and I want to use the specific preset values provided by the designer, which option do I choose?

**Comment by disconn3ct** [comment]
> Please read the description in the dialog, it's pretty clear already.

Please read the messages in the ticket, it is pretty clear that new users do not understand that dialog.

Now that that is out of the way, can we agree that the dialog phrasing is not perfect and could be improved?

**Comment by xTremePower** [comment]
I'm a long term user and it's still not clear at all ...

You lose everyone with the save/transfer things and also discard ...

If i'm on a project, and importe a 3mf with settings, 
What will happens if I Discard ? discard what ? the settings into the 3mf ?
What will happens if I SAVE ? Save into what ? into the profile actually on screen, or save the settings of the 3mf ?
What will happans if I Trasfer ? It will tranfer settings from the 3mf to into the actual settings ? in this case what the difference between save and transfer? 
If I already have the preset saved in my config settings, but I enabled one other one before opening the 3mf it will still create a new preset or use the saved one ?!

Many times I make button error, and I need to think about to push the orange rollback circles in printer, filamen and config windows before saving myself or I will trash my actual good profile .....

**Comment by MxBrnr** [comment]
Apparently I am the only person who has thought the implementation of this dialog has been completely self-explanatory from the beginning. 🤷 Admittedly, "Transfer" is better and less confusing than "Keep," but beyond that the 3 option are clearly different and seem quite obvious, with the wording being adequately consistent with what is commonly used in software and other digital applications.

**Comment by klwemu** [comment]
It's totally confusing and it is not enogh to change some text labels. 
The UI should be redesigned and to be more intuitive.

**Comment by PitBull440** [comment]
When you Transfer the changes to the newly selected preset, does it Save or Discard the changes to the preset that _was_ selected?

**Comment by MxBrnr** [comment]
Discards, because you did not save them to that profile before switching.

**Comment by nilesjohnson** [comment]
I don't think 2 min of self-experimentation clarifies this, nor do I think saying so is a helpful way to proceed.  

I found this issue while internet searching for some explanation.  I have a project where I've tweaked a bunch of settings just for those particular objects.  When I try to quit Orca Slicer, I'm presented with this dialog.  I'm not trying to change presets.  I don't want to lose the settings I've dialed in for this particular project, and I also don't want these settings being copied over to my other projects.  I can't imagine this is an unusual situation.

Also, please clarify about #9477: can this issue be reopened and that one closed as duplicate?

**Comment by MxBrnr** [comment]
> I don't think 2 min of self-experimentation clarifies this, nor do I think saying so is a helpful way to proceed.
> 
> I found this issue while internet searching for some explanation. I have a project where I've tweaked a bunch of settings just for those particular objects. When I try to quit Orca Slicer, I'm presented with this dialog. I'm not trying to change presets. I don't want to lose the settings I've dialed in for this particular project, and I also don't want these settings being copied over to my other projects. I can't imagine this is an unusual situation.
> 
> Also, please clarify about [#9477](https://github.com/SoftFever/OrcaSlicer/issues/9477): can this issue be reopened and that one closed as duplicate?

Then you need to save those settings as a new profile, or save the project 3MF, before changing or exiting

**Comment by nilesjohnson** [comment]
> Then you need to save those settings as a new profile, or save the project 3MF, before changing or exiting

Thanks.  If the dialog isn't going to change, how about adding a section on the wiki page so that anyone who is still confused can find extra explanation there?  A starting point would be [this earlier comment](https://github.com/SoftFever/OrcaSlicer/issues/4422#issuecomment-2618329228).

>This dialog popup when you switch presets and there are unsaved changes.
So these options are:
Transfer (changes to new preset)
Discard (changes)
Save changes into current preset then switch to new preset

While doing this, someone could clarify whether any of the buttons saves the current 3mf, or that has to be done separately.

Depending on what's best, I'd be happy to
- open a new ticket for this, 
- send a PR with draft changes for the wiki, and/or
- leave it to someone else who knows better how to do these things

**Comment by MxBrnr** [comment]
> While doing this, someone could clarify whether any of the buttons saves the current 3mf, or that has to be done separately.

The 3MF must be saved separately.

**Comment by alexandrejgoes** [comment]
This is the same root issue I just commented over at #11487. Might be worth merging the two threads since they're describing the same problem from slightly different angles.

-----

I strongly agree this dialog needs fixing. I ran into the exact same confusion described above. The root issue isn't just the button labels, it's that "Old Value" / "New Value" describes the wrong axis. Users instinctively read "New" as "more recent," but in practice "New Value" is whatever's coming from the loaded file, which is very often the older config (e.g. loading an old project after refining presets since).

Columns: rename to describe origin instead of recency, so the ambiguity disappears without even touching button wording:

"Old Value" => "Current Preset"
"New Value" => "From File"

Buttons: the current three options (Transfer / Discard / Save) also don't map cleanly onto what they do. Suggest collapsing to two, since "Save" (apply + persist to disk) is redundant. Once values are applied, the preset shows as modified and the user can save it normally via the existing preset-save control whenever they want:

"Keep Current" => keep the preset as-is, ignore the file's values.
"Use File" => apply the file's values to the working preset (marked modified, savable later as normal).

Both are verb + clear target, mirroring each other (Keep/Use, Current/File), so the action is obvious without needing to read the table first.

**Comment by LJKraus** [comment]
The various Transfer/Save/Discard dialogs have had many forms over the years, both in OrcaSlicer and Bambu Studio. Changing the wording has helped some users, but it then confuses others.

I'm convinced the problem is not the wording, but users not taking the time to actually read the stated purpose of the dialog, or who simply have poor reading comprehension.

**Comment by Raketemensch** [comment]
Good software design doesn't send users to forums or reddit or AI to understand what a dialog is asking them.
